app/login.go

```diff
@@ -55,6 +58,26 @@ func (a *App) AuthenticateUserForLogin(id, loginId, password, mfaToken string, l
 		return nil, err
 	}
 
+	// Cloud login allow to use the one-time token to
+	// login the user to their installation for the first time
+	if IsCWSLogin(a) {
+		if err = checkUserNotBot(user); err != nil {
+			return nil, err
+		}
+		sessions, errSessions := a.GetSessions(user.Id)
+		if errSessions != nil {
+			return nil, errSessions
+		}
+		if len(sessions) > 0 {
+			return nil, model.NewAppError("AuthenticateUserForLogin",
+				"api.user.login_by_cws.invalid_token.app_error", nil, "", http.StatusBadRequest)
+		}
+		token, ok := os.LookupEnv(cwsTokenEnv)
+		if ok && cwsToken != "" && token == cwsToken {
+			return user, nil
+		}
```

Comment:
Might be a good idea to check if the token is set first before checking if they're a bot, pulling the sessions, etc. otherwise we'll be doing all that extra work every time someone logs in normally, right?

Reply:
Yep you're right fixed checking the cws token in the `IsCWSLogin` method